content/articles/creating-a-pyspark-data-frame-a-beginners-guide/index.md

```diff
@@ -0,0 +1,159 @@
+```
+#### 2. Creating the spark session
+```python
+sparkSession = SparkSession.builder.appName('Creating PySpark dataframe from an external file').getOrCreate()
+```
+Using `appName(),` we can retrieve the application's name by passing in a string. It is possible to leverage an existing spark session or build a new one using the `getOrCreate()` method.
+#### 3. Read the external files into the PySpark data frame
+The external file format that we will use and import comprises a CSV file.
+
+```python
+ourCSV_file = sparkSession.read.csv('IncomeAndGender.csv', sep = ',', inferSchema = True, header = True)
+```
+Delimiters for the CSV file were given (which is a comma). Afterward, the inferSchema property was set to True. CSV files will be automatically transformed into PySpark data frames with this command. 
+
+Let's have a look at our PySpark data frame's data kinds now:
+```python
+print(type(ourCSV_file))
+```
+Output:
+```bash
+<class 'pyspark.sql.dataframe.DataFrame'>
+```
+We can also check the schema:
+```python
+ourCSV_file.printSchema()
+```
+Output:
+```bash
+root
+ |-- Salary: double (nullable = true)
+ |-- Gender: integer (nullable = true)
+ |-- AgeNumber: double (nullable = true)
+ |-- PhD: string (nullable = true)
+```
+> To see the complete code for this [click here](https://colab.research.google.com/drive/1u04j9muZ1FdDvr2oxeuSGST5Lva9es9j?usp=sharing). In addition, the CSV file that we used here was initially created in our notebook.
+
+### Other methods 
+#### 1. Converting PySpark data frame to a Pandas data frame
+The PySpark data frame can be converted to a Pandas data frame, and Pandas techniques can be used on our data frame, which is incredibly convenient.
+```python
+csvToPandas = ourCSV_file.toPandas()
+```
+To ensure that everything we're doing is correct, let's look at the data type of our new data frame.
+```python
+type(csvToPandas)
+```
+Output:
+```bash
+pandas.core.frame.DataFrame
+```
+#### 2. Reading multiple files
+In the `read()` method, we can pass a list of file paths as a string type, creating and allocating a PySpark data frame in the variable `csvToPandas.` A shared data frame should not be deemed significant while reading several files at the same time.
+### Reference
+1. See full code for [creating PySpark data frame from an external file](https://colab.research.google.com/drive/1u04j9muZ1FdDvr2oxeuSGST5Lva9es9j?usp=sharing).
+2. See whole code for [creating a PySpark data frame using an existing Resilient Distributed Dataset](https://colab.research.google.com/drive/1Ykwifphdj2mLQ9rhKuX309aXq-tGurTD?usp=sharing)
+3. [pyspark.sql.DataFrame](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.sql.DataFrame.html)
+### Conclusion
+Compared to `read()` methods, `parallelize()` is better, but it necessitates more work. When we need to fast scan a CSV file, the `read()` methods are helpful. Many options are available when making changes to a PySpark data frame.
```

Comment:
Use the conclusion to summarize the main ideas that you explained through this article, add more value to it.